huaweicloud/services/lb/resource_huaweicloud_lb_pool.go

```diff
@@ -145,31 +145,11 @@ func resourcePoolV2Create(ctx context.Context, d *schema.ResourceData, meta inte
 	adminStateUp := d.Get("admin_state_up").(bool)
 	var persistence pools.SessionPersistence
 	if p, ok := d.GetOk("persistence"); ok {
-		pV := (p.([]interface{}))[0].(map[string]interface{})
-
-		persistence = pools.SessionPersistence{
-			Type: pV["type"].(string),
-		}
-
-		if persistence.Type == "APP_COOKIE" {
-			if pV["cookie_name"].(string) == "" {
-				return fmtp.DiagErrorf(
-					"Persistence cookie_name needs to be set if using 'APP_COOKIE' persistence type")
-			}
-			persistence.CookieName = pV["cookie_name"].(string)
-
-			if pV["timeout"].(int) != 0 {
-				return fmtp.DiagErrorf(
-					"Persistence timeout is invalid when type is set to 'APP_COOKIE'")
-			}
-		} else {
-			if pV["cookie_name"].(string) != "" {
-				return fmtp.DiagErrorf(
-					"Persistence cookie_name can only be set if using 'APP_COOKIE' persistence type")
-			}
+		persistenceObj, buildPersistenceErr := buildPersistence(p)
+		if buildPersistenceErr != nil {
+			return buildPersistenceErr
 		}
-
-		persistence.PersistenceTimeout = pV["timeout"].(int)
+		persistence = persistenceObj
```

Comment:
ditto

Reply:
done

Reply:
we don't need to change this file because the `timeout` field was defined in the LB resource